builtin/logical/aws/secret_access_keys.go

```diff
@@ -67,14 +68,20 @@ func genUsername(displayName, policyName, userType string) (ret string, warning
 func (b *backend) secretTokenCreate(ctx context.Context, s logical.Storage,
 	displayName, policyName, policy string,
 	lifeTimeInSeconds int64) (*logical.Response, error) {
-	STSClient, err := clientSTS(ctx, s)
-	if err != nil {
-		return logical.ErrorResponse(err.Error()), nil
+
+	b.clientMutex.Lock()
```

Comment:
Consider making `clientMutex` an RWLock that can be upgraded from RLock to Lock if client is nil. This avoids synchronizing all of the read requests. You can see an example upgrade pattern (which shows up in various place in Vault) here:  https://github.com/hashicorp/vault-plugin-secrets-azure/blob/master/client.go#L37